packages/sidecar/src/utils/docker-compose-parser.ts

```diff
@@ -2,40 +2,49 @@ import { existsSync, readFileSync } from "node:fs";
 import { parse as parseYaml } from "yaml";
 import { logger } from "../logger.js";
 
-const COMPOSE_FILE_NAMES = ["docker-compose.yaml", "docker-compose.yml", "compose.yml", "compose.yaml"] as const;
+const COMPOSE_FILE_BASE_NAMES = ["docker-compose", "compose"] as const;
+const COMPOSE_FILE_EXTENSIONS = [".yml", ".yaml"] as const;
 
 /**
- * Find the compose file in the current directory
+ * Find a file by trying all combinations of base names and extensions.
  */
-export function findComposeFile(): string | null {
-  for (const fileName of COMPOSE_FILE_NAMES) {
-    if (existsSync(fileName)) {
-      return fileName;
+function findFile(buildFileName: (baseName: string, ext: string) => string): string | null {
+  for (const baseName of COMPOSE_FILE_BASE_NAMES) {
+    for (const ext of COMPOSE_FILE_EXTENSIONS) {
+      const fileName = buildFileName(baseName, ext);
+      if (existsSync(fileName)) {
+        return fileName;
+      }
     }
   }
   return null;
 }
 
+/**
+ * Find the compose file in the current directory
+ */
+export function findComposeFile(): string | null {
+  return findFile((baseName, ext) => `${baseName}${ext}`);
+}
+
 /**
  * Find the override file for a given compose file
  */
 export function findOverrideFile(composeFile: string): string | null {
   const ext = composeFile.endsWith(".yaml") ? ".yaml" : ".yml";
   const baseName = composeFile.replace(/\.(yaml|yml)$/, "");
-
   const overrideFile = `${baseName}.override${ext}`;
 
   if (existsSync(overrideFile)) {
     return overrideFile;
   }
-
   return null;
 }
```

Comment:
I think this needs to use `findFile` too? Otherwise `findFile` taking a builder function as a parameter makes no sense to me.